rustls/src/client/tls12.rs

```diff
@@ -822,7 +845,7 @@ impl hs::State for ExpectFinished {
         // Hash this message too.
         st.handshake.transcript.add_message(&m);
 
-        save_session(&st.secrets, &mut st.handshake, &mut st.ticket, sess);
+        save_session(&st.secrets, &mut st.handshake, &mut st.ticket, sess, st.negotiated_version);
```

Comment:
I know it looks messy to thread `negotiated_version` through to this point just for saving the session, but I think in the near future we'll improve it so that what is currently in `st.ticket`, `st.handshake.session_id`, `st.negotiated_version`, and maybe more will all be held in a single `SessionDataToSave` structure that gets threaded through. So, I'm happy to see this incremental step done.